lib/Sabberworm/CSS/Value/Size.php

```diff
@@ -36,11 +37,33 @@ public static function parse(ParserState $oParserState, $bIsColorComponent = fal
 			}
 		}
 
+		$sParsedUnit = '';
+		$iOffset = 0;
+		while (true) {
+			$sChar = $oParserState->peek(1, $iOffset);
+			$iPeek = ord($sChar);
+
+			// Ranges: a-z A-Z 0-9 %
+			if (($iPeek >= 97 && $iPeek <= 122) ||
+				($iPeek >= 65 && $iPeek <= 90) ||
+				($iPeek >= 48 && $iPeek <= 57) ||
+				($iPeek === 37)) {
+				$sParsedUnit .= $sChar;
+				$iOffset++;
+			} else {
+				break;
+			}
+		}
```

Comment:
I'm not entirely comfortable with this approach of getting the unit; better solutions are welcomed.

Reply:
Instead of a `while` loop, another option would be to peek 10 chars and then use `preg_match()` to match the range, and if matched then capture that as the `$sParsedUnit` and increase the `$iOffset` by the length.

```
if ( preg_match( '/^[a-zA-Z0-9%]+/', $oParserState->peek(10, $iOffset), $matches ) ) {
    $sParsedUnit = $matches[0];
    $iOffset += strlen( $matches[0] );
}
```

Reply:
Any reason behind the 10 characters? Or is it that a unit length of 10 characters or more would be unlikely?

Reply:
Good point. It could be set to the max length of the unit chars.